nucypher/blockchain/eth/sol/source/contracts/PolicyManager.sol

```diff
@@ -291,50 +291,66 @@ contract PolicyManager is Upgradeable {
     }
 
     /**
-    * @notice Set default `feeDelta` value for specified period
-    * @dev This method increases gas cost for node in trade of decreasing cost for policy sponsor
+    * @notice Call from StakingEscrow to update node info once per period.
+    * Set default `feeDelta` value for specified period and update node fee
     * @param _node Node address
-    * @param _period Period to set
-    */
-    function setDefaultFeeDelta(address _node, uint16 _period) external onlyEscrowContract {
+    * @param _processedPeriod1 Processed period
+    * @param _processedPeriod2 Processed period
+    * @param _periodToSetDefault Period to set
+    */
+    function ping(
+        address _node,
+        uint16 _processedPeriod1,
+        uint16 _processedPeriod2,
+        uint16 _periodToSetDefault
+    )
+        external onlyEscrowContract
+    {
         NodeInfo storage node = nodes[_node];
-        if (node.feeDelta[_period] == 0) {
-            node.feeDelta[_period] = DEFAULT_FEE_DELTA;
+        if (_processedPeriod1 != 0) {
+            updateFee(_node, node, _processedPeriod1);
+        }
+        if (_processedPeriod2 != 0) {
+            updateFee(_node, node, _processedPeriod2);
+        }
+        // This code increases gas cost for node in trade of decreasing cost for policy sponsor
+        if (_periodToSetDefault != 0 && node.feeDelta[_periodToSetDefault] == 0) {
+            node.feeDelta[_periodToSetDefault] = DEFAULT_FEE_DELTA;
         }
     }
 
     /**
     * @notice Update node fee
     * @param _node Node address
+    * @param _info Node info structure
     * @param _period Processed period
     */
-    function updateFee(address _node, uint16 _period) external onlyEscrowContract {
-        NodeInfo storage node = nodes[_node];
-        if (node.previousFeePeriod == 0 || _period <= node.previousFeePeriod) {
+    function updateFee(address _node, NodeInfo storage _info, uint16 _period) internal {
```

Comment:
It seems the `_node` parameter is not used now that we have the `_info` struct, except for the `NodeBrokenStake` event... 
BTW, when did we use that event?

Reply:
It was for bug detection (while one of testnets), it should never emit this event, and event here only because of paranoia. If you think - we can remove it

Reply:
Yes, I think we can remove it.

Reply:
✔️